src/Sylius/Bundle/CoreBundle/Fixture/Listener/CatalogPromotionExecutorListener.php

```diff
@@ -15,20 +15,31 @@
 
 use Sylius\Bundle\CoreBundle\Fixture\CatalogPromotionFixture;
 use Sylius\Bundle\CoreBundle\Processor\AllProductVariantsCatalogPromotionsProcessorInterface;
+use Sylius\Bundle\CoreBundle\Processor\CatalogPromotionStateProcessorInterface;
 use Sylius\Bundle\FixturesBundle\Listener\AbstractListener;
 use Sylius\Bundle\FixturesBundle\Listener\AfterFixtureListenerInterface;
 use Sylius\Bundle\FixturesBundle\Listener\FixtureEvent;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
 
 final class CatalogPromotionExecutorListener extends AbstractListener implements AfterFixtureListenerInterface
 {
-    public function __construct(private AllProductVariantsCatalogPromotionsProcessorInterface $allCatalogPromotionsProcessor)
-    {
+    public function __construct(
+        private AllProductVariantsCatalogPromotionsProcessorInterface $allCatalogPromotionsProcessor,
+        private CatalogPromotionStateProcessorInterface $catalogPromotionStateProcessor,
+        private RepositoryInterface $catalogPromotionsRepository
+    ) {
     }
 
     public function afterFixture(FixtureEvent $fixtureEvent, array $options): void
     {
         if ($fixtureEvent->fixture() instanceof CatalogPromotionFixture) {
             $this->allCatalogPromotionsProcessor->process();
+
+            $catalogPromotions = $this->catalogPromotionsRepository->findAll();
+
+            foreach ($catalogPromotions as $catalogPromotion) {
+                $this->catalogPromotionStateProcessor->process($catalogPromotion);
+            }
```

Comment:
I wonder if we shouldn't connect state processor with catalog promotions processor, it seems to be quite logical, but it could be done in a separate PR.

Probably we can improve this code by finding only eligible catalog promotions here:
```suggestion
            $catalogPromotions = $this->catalogPromotionsRepository->findByCriteria(...);

            foreach ($catalogPromotions as $catalogPromotion) {
                $this->catalogPromotionStateProcessor->process($catalogPromotion);
            }
```